ibis/udf/validate.py

```diff
@@ -0,0 +1,71 @@
+"""Validation for UDFs.
+
+Warning: This is an experimental module and API here can change without notice.
+
+DO NOT USE DIRECTLY.
+"""
+
+from inspect import Parameter, signature
+
+import ibis.common.exceptions as com
+
+
+def _parameter_count(funcsig):
```

Comment:
Why take this out? The origin file is not very large.

Reply:
I think this depends on whether we want to do this for vectorized UDFs only or all backends. I moved this out so that the import wouldn't be `ibis.udf.vectorized` the functions in here

Reply:
(see https://github.com/ibis-project/ibis/pull/2198#discussion_r432728880)